cajita/src/Cajita_FastFourierTransform.hpp

```diff
@@ -23,291 +23,509 @@
 #include <memory>
 #include <type_traits>
 
+#ifdef KOKKOS_ENABLE_HIP // FIXME_HIP
+#error FFT not supported with Kokkos::Experimental::HIP backend
+#endif
+
 namespace Cajita
 {
 namespace Experimental
 {
 //---------------------------------------------------------------------------//
-template <class MemorySpace>
-struct HeffteMemoryTraits;
 
+/*!
+  \class FFTScaleFull
+  \brief Tag for full scaling of FFT.
+*/
+struct FFTScaleFull
+{
+};
+/*!
+  \class FFTScaleNone
+  \brief Tag for no scaling of FFT.
+*/
+struct FFTScaleNone
+{
+};
+/*!
+  \class FFTScaleSymmetric
+  \brief Tag for symmetric scaling of FFT.
+*/
+struct FFTScaleSymmetric
+{
+};
+
+/*!
+  \class FFTBackendFFTW
+  \brief Tag specifying FFTW backend for FFT (host default).
+*/
+struct FFTBackendFFTW
```

Comment:
These backend options are heFFTe specific. All backend stuff should be put in some kind of heFFTe namespace or in the heFFTe class.

Reply:
So I would put them in the `Impl` namespace below and prefix all the struct names with `Heffte`.

Reply:
If we find that other FFT libraries do the same thing we can relax this and generalize the backend concept to some kind of 1D FFT concept.

Reply:
I left them for now because the same tags could be used for the other FFT libraries. The only issue there is disabling backends that don't exist for a given library (fftMPI has one more option than heFFTe)